src/Controller/Async/Oembed.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace Bolt\Controller\Async;
+
+use Embed\Exceptions\InvalidUrlException;
+use Embed\Http\Url;
+use Silex\ControllerCollection;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Async controller for oEmbed routes.
+ *
+ * @author Gawain Lynch <gawain.lynch@gmail.com>
+ */
+class Oembed extends AsyncBase
+{
+    /**
+     * {@inheritdoc}
+     */
+    protected function addRoutes(ControllerCollection $c)
+    {
+        $c->post('/oembed/request', 'oEmbedRequestEndpoint')
+            ->bind('oEmbedRequestEndpoint')
+        ;
+    }
+
+    /**
+     * @param Request $request
+     *
+     * @throws InvalidUrlException
+     *
+     * @return JsonResponse
+     */
+    public function oEmbedRequestEndpoint(Request $request)
+    {
+        if (!$this->isCsrfTokenValid($request->request->get('bolt_csrf_token'))) {
+            return new JsonResponse(['error' => ['message' => 'Invalid CSRF token']], Response::HTTP_FORBIDDEN);
+        }
+        if (!$request->request->get('has')) {
+            return new JsonResponse(['error' => ['message' => 'Invalid POST parameters']], Response::HTTP_FORBIDDEN);
+        }
+        $url = $request->request->get('url');
+        $url = Url::create($url);
+
+        try {
+            $info = $this->getUrlAdapter($url);
+        } catch (InvalidUrlException $e) {
+            $response = ['error' => ['message' => $e->getMessage()]];
+
+            return new JsonResponse($response, Response::HTTP_SERVICE_UNAVAILABLE);
+        }
+        /** @var \Embed\Providers\Provider[] $providers */
+        $providers = $info->getProviders();
+        /** @var \Embed\Providers\OEmbed $oEmbed */
+        $oEmbed = $providers['oembed'];
+
+        return new JsonResponse($oEmbed->getBag()->getAll());
+    }
+
+    /**
+     * Return an API adapter matching the pattern of the given URL.
+     *
+     * @param Url $url
+     *
+     * @return \Embed\Adapters\Adapter
+     */
+    protected function getUrlAdapter(Url $url)
+    {
+        $oEmbedFactory = $this->app['oembed.factory'];
+
+        return $oEmbedFactory($url);
+    }
```

Comment:
I kinda think having our own service class that wraps the ombed api that the controller could call here would be better. Then the controller doesn't have to know anything about the `Embed\*` code. But if there's a rush don't worry about it.